sound/soc/sof/topology.c

```diff
@@ -2002,6 +2011,10 @@ static int sof_manifest(struct snd_soc_component *scomp, int index,
 		}
 	}
 
+	/* get nhlt from bios if exists and no override set */
+	if (!sof_use_tplg_nhlt)
+		sdev->nhlt = intel_nhlt_init(sdev->dev);
```

Comment:
this makes me cringe. The intel NHTL stuff should be Intel-specific, I don't think it's wise to have this code in the SOF core.

Reply:
@plbossart makes sense. We will move it to the intel-specific code where we do the check for DMIC's